src/highdicom/content.py

```diff
@@ -936,3 +1067,536 @@ def __init__(
+                self.WindowWidth = [
+                    format_number_as_ds(x) for x in window_width
+                ]
+            else:
+                if window_is_sequence:
+                    raise TypeError(
+                        'Length of "window_width" must match length of '
+                        '"window_center".'
+                    )
+                self.WindowWidth = format_number_as_ds(window_width)
+        if window_explanation is not None:
+            if window_center is None:
+                raise TypeError(
+                    'Providing "window_explanation" is invalid if '
+                    '"window_center" is not provided.'
+                )
+            if isinstance(window_explanation, str):
+                if window_is_sequence:
+                    raise TypeError(
+                        'Length of "window_explanation" must match length of '
+                        '"window_center".'
+                    )
+                _check_long_string(window_explanation)
+            elif isinstance(window_explanation, Sequence):
+                if (
+                    not window_is_sequence or
+                    (len(window_explanation) != len(window_center))
+                ):
+                    raise ValueError(
+                        'Length of "window_explanation" must match length of '
+                        '"window_center".'
+                    )
+                if len(window_explanation) == 0:
+                    raise TypeError(
+                        'Argument "window_explanation" must not be an empty '
+                        'sequence.'
+                    )
+                for exp in window_explanation:
+                    _check_long_string(exp)
+            self.WindowCenterWidthExplanation = window_explanation
+        if voi_lut_function is not None:
+            if window_center is None:
+                raise TypeError(
+                    'Providing "voi_lut_function" is invalid if '
+                    '"window_center" is not provided.'
+                )
+            self.VOILUTFunction = VOILUTFunctionValues(voi_lut_function).value
+
+        if voi_luts is not None:
+            if len(voi_luts) == 0:
+                raise ValueError('"voi_luts" should not be empty.')
+            for lut in voi_luts:
+                if not isinstance(lut, LUT):
+                    raise TypeError(
+                        'Items of "voi_luts" should be of type highdicom.LUT.'
+                    )
+            self.VOILUTSequence = list(voi_luts)
+        else:
+            if window_center is None:
+                raise TypeError(
+                    'At least one of "window_center" or "voi_luts" should be '
+                    'provided.'
+                )
+
+
+class PaletteColorLookupTable(Dataset):
+
+    """Dataset describing a palette color lookup table."""
+
+    def __init__(
+        self,
+        red_palette_color_lut_data: np.ndarray,
```

Comment:
What about segmented LUT data?

Reply:
I have not implemented the Segmented Palette Color Lookup Table data. This seems pretty specialist and I do not see an immediate use case, plus it seems rather fiddly. I think this could be added in the future by writing a SegmentedLUT class, and changing the types of the parameters of PaletteColorLookupTable from LUT to Union[LUT, SegmentedLUT]. What do you think?

The "usual" palette color LUTs (implemented) and the Segmented LUTs are mutually exclusive

Reply:
The segmented LUTs are quite important in practice, because they may safe a lot of space - in particular with 16-bit LUTs. Consequently, the data may then also be considered metadata rather than bulkdata, which can significantly simply the workflow on display devices, since they do not need to fetch the bulkdata via separate requests. Most colormaps used in practice can probably be build from a few segments using interpolation.

Reply:
You may want to take a look at the [JavaScript version](https://github.com/herrmannlab/dicom-microscopy-viewer/blob/5dea96a647b95f9775d4b79e8895f88d8fefb241/src/color.js#L119) for inspiration